### PR TITLE
[InstCombine] Swap out range metadata to range attribute for arm_mve_pred_v2i

### PR DESCRIPTION
**Pull request by andjo403**

### CODE REVIEW

**Comment by andjo403** [comment]
Continues to swap out range metadata to range attribute for calls to be able to deprecate range metadata on calls in the future

File: llvm/lib/CodeGen/SelectionDAG/SelectionDAGBuilder.cpp [hunk 1]
```diff
@@ -10227,21 +10227,25 @@ void SelectionDAGBuilder::visitVACopy(const CallInst &I) {
 }
 
 SDValue SelectionDAGBuilder::lowerRangeToAssertZExt(SelectionDAG &DAG,
-                                                    const Instruction &I,
```
Comment (arsenm) [inline llvm/lib/CodeGen/SelectionDAG/SelectionDAGBuilder.cpp:10230]: This should not lose the ability to handle range metadata. We should still be able to lower loads etc. with range to include the assert

Comment (andjo403) [reply llvm/lib/CodeGen/SelectionDAG/SelectionDAGBuilder.cpp:10230]: ok will fix, only noticed that it was called for calls now and it made the code a bit simpler.

**Comment by nikic** [comment]
> > to be able to deprecate range metadata on calls in the future
> 
> Meaning what exactly? I don't think calls should be special cased to disallow range metadata, it would just be a less canonical form?

We generally require that calls use return attributes instead of metadata. E.g. the verifier will reject !nonnull or !align on a call, you have to use the nonnull or align attribute instead. We'll do the same for !range metadata vs attribute once it's fully migrated.

**Comment by andjo403** [comment]
about the deprecation it was my interpretation of the comment from @nikic here 
https://github.com/llvm/llvm-project/pull/84627#discussion_r1518589822 but deprecate as in the Verifier not allow range metadata on calls and invokes

File: llvm/lib/CodeGen/SelectionDAG/SelectionDAGBuilder.cpp (hunk 1 above)
Comment (andjo403) [reply llvm/lib/CodeGen/SelectionDAG/SelectionDAGBuilder.cpp:10230]: this is fixed now

File: llvm/lib/CodeGen/SelectionDAG/SelectionDAGBuilder.cpp [hunk 2]
```diff
@@ -4487,6 +4487,18 @@ static const MDNode *getRangeMetadata(const Instruction &I) {
   return I.getMetadata(LLVMContext::MD_range);
 }
 
+static std::optional<ConstantRange> getRange(const Instruction &I) {
```
Comment (nikic) [inline llvm/lib/CodeGen/SelectionDAG/SelectionDAGBuilder.cpp:4490]: Could you please split out the SDAG change into a separate PR?

**Comment by andjo403** [comment]
This is rebased on the part that was split out now and ready for review

**Review by nikic — APPROVED** [review]
LGTM
